File: spring-cloud-integrated-demo/user-service/user-service-api/src/main/java/com/github/jccode/springcloud/integrateddemo/user/api/UserAPIFallback.java
package com.github.jccode.springcloud.integrateddemo.user.api;

import com.github.jccode.springbootsample.core.data.rest.RestResult;
import com.github.jccode.springcloud.integrateddemo.user.model.User;
import org.springframework.stereotype.Component;

import static com.github.jccode.springbootsample.core.utils.CommonUtil.fail;

@Component
public class UserAPIFallback implements UserAPI {

    @Override
    public RestResult<User> findByName(String name) {
        return fail("user service is not available");
    }

    @Override
    public RestResult<User> find(Integer id) {
        return fail("user service is not available");
    }
}

File: spring-cloud-integrated-demo/user-service/user-service-api/src/main/scala/com/github/jccode/springcloud/integrateddemo/user/api/UserApiFallback.scala
package com.github.jccode.springcloud.integrateddemo.user.api

import com.github.jccode.springbootsample.core.data.RestResult
import com.github.jccode.springbootsample.core.data.RestResult.RestSuccess
import com.github.jccode.springcloud.integrateddemo.user.model.User
import org.springframework.stereotype.Component

@Component
class UserApiFallback extends UserApi {

// TODO 这里能获取业务抛出的异常吗?
  override def findByName(name: String): RestSuccess[User] = RestResult.fail("user service is not available")

  override def find(id: Integer): RestSuccess[User] = RestResult.fail("user service is not available")
}
